client/client.go

```diff
@@ -231,10 +233,10 @@ func (c *client) tsLoop() {
 
 		select {
 		case first := <-c.tsoRequests:
-			requests = append(requests, first)
-			pending := len(c.tsoRequests)
-			for i := 0; i < pending; i++ {
-				requests = append(requests, <-c.tsoRequests)
+			pendingPlus1 := len(c.tsoRequests) + 1
```

Comment:
`pendingPlus1` can be greater than `defaultSize`.

Reply:
max value for `len(c.tsoRequests)` can be `maxMergeTSORequests`.
Below for loop will be iterating `pendingPlus1 - 1` times as starting index is 1 instead of 0.
Max index value for `i` can be `pendingPlus1 - 1` which can be maximum `maxMergeTSORequests`.
`defaultSize` is set to `maxMergeTSORequests + 1` so it will allow array index to be `maxMergeTSORequests`.
So, `pendingPlus1` will never be greater than `defaultSize`.
https://play.golang.org/p/1kEQsGbFdnN will help in testing above scenario.

Reply:
cool, I think I didn't realize the channel is a buffered channel. 😅